scripts/polyrith_sage.py

```diff
@@ -0,0 +1,87 @@
+# This file is part of the `polyrith` tactic in `src/tactic/polyrith.lean`.
+# It interfaces between Lean and the Sage web interface.
+
+import requests
+import json
+import sys
+from os.path import join, dirname
+
+# These functions are used to format the output of Sage for parsing in Lean.
+# They are stored here as a string since they are passed to Sage via the web API.
+with open(join(dirname(__file__), "polyrith_sage_helper.py"), encoding='utf8') as f:
+    polynomial_formatting_functions = f.read()
+
+# future extensions may change behavior depending on the base type
+def type_str(type):
+    return "QQ"
+
+def create_query(type: str, n_vars: int, eq_list, goal_type):
+    """ Create a query to invoke Sage's `MPolynomial_libsingular.lift`. See
+    https://github.com/sagemath/sage/blob/f8df80820dc7321dc9b18c9644c3b8315999670b/src/sage/rings/polynomial/multi_polynomial_libsingular.pyx#L4472-L4518
+    for a description of this method. """
+    var_list = ", ".join([f"var{i}" for i in range(n_vars)])
+    query = f'''
+import json
+P = PolynomialRing(QQ, 'var', {n_vars!r})
+[{var_list}] = P.gens()
+gens = {eq_list}
+p = P{goal_type}
```

Comment:
```suggestion
p = P({goal_type})
```
makes it more obvious what's going on here